e3sm_diags/metrics/metrics.py

```diff
@@ -0,0 +1,233 @@
+"""This module stores functions to calculate metrics using Xarray objects."""
+from __future__ import annotations
+
+from typing import List
+
+import xarray as xr
+import xcdat as xc
+import xskillscore as xs
+
+from e3sm_diags.logger import custom_logger
+
+logger = custom_logger(__name__)
+
+AXES = ["X", "Y"]
+
+
+def get_weights(ds: xr.Dataset):
```

Comment:
There is minor code inconsistency in computing weighted metrics, example, `std` computes weights in its own function, while `correlation`, `rmse` has weights passed in. I suggest to have weights computed in each metrics to be consistent. This means to compute weights one time more (?), but reduces the number of functions (`get_weights`) and arguments(`weights`)  being passed around..

Reply:
yes. The assumption is that both ds_a and ds_b should have the same grids/ area weights.

Reply:
Thanks, I updated these functions in [`a19e9a4` (#677)](https://github.com/E3SM-Project/e3sm_diags/pull/677/commits/a19e9a4148b8ba18135104aae7ed80cc5a67e02d#diff-3f01f708754161d2742b9affdeec02a41fef2e6f027c3175051262e06aedaad7). I also removed the `serialize` argument because we always serialize numpy arrays to a list of floats before writing them to`.json`.